lib/r10k/util/purgeable.rb

```diff
@@ -9,6 +9,16 @@ module Util
     #   {#desired_contents}
     module Purgeable
 
+      # Matches a directory path that contains a directory that starts with a dot '.'.
+      # Directory must start with a dot and then have a non-dot character, so the *nix
+      # special directories '.', and '..' are invalid. Matching path must have the
+      # dot-prepended segment as an interior segment (not at the end of the path).
+      HIDDEN_SUBPATH = Regexp.new("#{File::SEPARATOR}\\.[^.]+#{File::SEPARATOR}.*")
```

Comment:
Why does `FNM_DOTMATCH` not cover this?

Reply:
The difference in behavior stemmed from them not passing `FNM_DOTMATCH` into `Dir.glob` previously. I've updated the code this morning, which I _think_ is less convoluted. Let me know what you think!